### PR TITLE
stm32f4: Add STM32F429I-DISC1 board

### PR DESCRIPTION
**Pull request by jli157**
Add necessary board files, pinmux and device tree in order to have a
usable debug console.

ZEP-2461

Signed-off-by: Jun Li <jun.r.li@intel.com>

### CODE REVIEW

**Comment by sdiie** [comment]
Verified on samples/hello_world can be compiled with the option BOARD=stm32f429i_disc1 and can run on a STM32F429I-DISC1 board.

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 1]
```diff
@@ -0,0 +1,203 @@
+.. _stm32f429i_disc1_board:
+
+ST STM32F429I Discovery
+#######################
+
+Overview
+********
+
+The STM32F429I Discovery kit features an ARM Cortex-M4 based STM32F429ZI MCU
+with a wide range of connectivity support and configurations Here are
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: add a period after "configurations".

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	- Two user LEDs: LD3 (green), LD4 (red)
+	- Two USB OTG LEDs: LD5 (green) VBUS and LD6 (red) OC (over-current)
+
+.. image:: img/en.stm32f429i_disc1.png
+     :width: 350px
+     :align: center
+     :height: 631px
+     :alt: STM32F429I-DISC1
+
+More information about the board can be found at the `STM32F429IDISCOVERY website`_.
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: link name not found

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+- USB 2.0 OTG HS/FS with dedicated DMA, on-chip full-speed PHY and ULPI
+- 10/100 Ethernet MAC with dedicated DMA
+- 8- to 14-bit parallel camera
+- CRC calculation unit
+- True random number generator
+- DMA Controller
+
+More information about STM32F429ZI can be found here:
+       - `STM32F429ZI on www.st.com`_
+       - `STM32F429 reference manual`_
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: link name not found

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+	``boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig``
+
+
+Pin Mapping
+===========
+
+STM32F429I-DISC1 Discovry kit has 8 GPIO controllers. These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `STM32F429-DISC1 board User Manual`_.
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: link name not found

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+- DMA Controller
+
+More information about STM32F429ZI can be found here:
+       - `STM32F429ZI on www.st.com`_
+       - `STM32F429 reference manual`_
+
+Supported Features
+==================
+
+The Zephyr stm32f429i_disc1 board configuration supports the following hardware features:
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: You should be consistent in how you write stm32f429i_disc1 board throughout the doc (some references are all upper case, some without the underscore).

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+- UART_2_TX : PA2
+- UART_2_RX : PA3
+- USER_PB : PA0
+- LD3 : PG13
+- LD4 : PG12
+
+System Clock
+============
+
+STM32F429I-DISC1 System Clock could be driven by internal or external oscillator,
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: Could use some articles ("the", "an"):

     The STM32F429I-DISC1 system clock can be driven by an internal or external
     oscillator, as well as by the main PLL clock. By default the system clock is
     driven by the PLL clock at 168MHz, driven by an 8MHz high speed external
     clock.

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+as well as main PLL clock. By default System clock is driven by PLL clock at 168MHz,
+driven by 8MHz high speed external clock.
+
+Serial Port
+===========
+
+STM32F429I-DISC1 Discovery kit has up to 8 UARTs. The Zephyr console output is assigned to UART1.
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: The STM32...

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+Default settings are 115200 8N1.
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: Default communication settings are...

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+STM32F429I-DISC1 Discovery kit includes an ST-LINK/V2-B embedded debug tool interface.
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: The STM32...

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+This interface is supported by the openocd version included in Zephyr SDK.
+
+Flashing an application to STM32F4DISCOVERY
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: You're using the board name STM32F4DISCOVERY here, but elsewhere you're using STM32F429I-DISC1  (should be consistent).

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+   $ make BOARD=stm32f429i_disc1 flash
+
+Run a serial host program to connect with your board:
+
+.. code-block:: console
+
+   $ minicom -D /dev/ttyACM0
+
+You should see the following message:
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: Reset the board and you should see the following message:

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+.. _STM32F429I-DISC1 website:
+   http://www.st.com/en/evaluation-tools/32f429idiscovery.html
+
+.. _STM32F429I-DISC1 board User Manual:
+   http://www.st.com/web/en/resource/technical/document/user_manual/DM00097320.pdf
+
+.. _STM32F429ZI on www.st.com:
+   http://www.st.com/en/microcontrollers/stm32f429-439.html
+
+..
```
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: looks like a stray ".." at the end of the file (you're missing three link targets referenced earlier in the doc)

**Review by dbkinder — CHANGES_REQUESTED** [review]
some edits requested

**Comment by sdiie** [comment]
@dbkinder Please help to review again. I've already fixed all reported issues on the board document. Thank you!

File: boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst (hunk 11 above)
Comment (dbkinder) [inline boards/arm/stm32f429i_disc1/doc/stm32f429i_disc1.rst]: Missed this edit

**Review by dbkinder — CHANGES_REQUESTED** [review]
one missed edit, line 177, reset the board

**Review by dbkinder — DISMISSED** [review]
thanks!

**Comment by sdiie** [comment]
@dbkinder, thank you!

File: drivers/pinmux/stm32/pinmux_board_stm32f429i_disc1.c [hunk 13]
```diff
@@ -0,0 +1,38 @@
+/*
```
Comment (galak) [inline drivers/pinmux/stm32/pinmux_board_stm32f429i_disc1.c]: move the pinmux into boards/arm/stm32f429i_disc1

File: dts/arm/st/stm32f429.dtsi [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/stm32f407.dtsi>
+
+/ {
+	soc {
+		pinctrl: pin-controller {
```
Comment (galak) [inline dts/arm/st/stm32f429.dtsi]: you shouldn't need this pinctrl node, should be handled by what's in st/stm32f407.dtsi

Comment (sdiie) [inline dts/arm/st/stm32f429.dtsi]: Okay, will do.

File: drivers/pinmux/stm32/pinmux_board_stm32f429i_disc1.c (hunk 13 above)
Comment (sdiie) [inline drivers/pinmux/stm32/pinmux_board_stm32f429i_disc1.c]: @galak However, I see all ST Micro's board pinmux files are here. Is there a decision to move each of them back to each board directory?

File: boards/arm/stm32f429i_disc1/Makefile [hunk 15]
```diff
@@ -0,0 +1,10 @@
+# Makefile - Olimex STM32-E407 board
+#
+# Copyright (c) 2017, Erwin Rol <erwin@erwinrol.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+ccflags-y +=-I$(srctree)/drivers
```
Comment (galak) [inline boards/arm/stm32f429i_disc1/Makefile]: nit space after +=

**Review by galak — CHANGES_REQUESTED** [review]
Looks good, can you merge down to one commit.

**Comment by sdiie** [comment]
@galak Squashed four commits into one after adding amending the Makefile. Thank you!

**Review by galak — APPROVED** [review]

**Pull request merged**